src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -218,14 +230,7 @@ private SecurityStatusPal PrivateDecryptData(byte[]? buffer, ref int offset, ref
         private async Task ReplyOnReAuthenticationAsync<TIOAdapter>(TIOAdapter adapter, byte[]? buffer)
             where TIOAdapter : ISslIOAdapter
         {
-            lock (SyncLock!)
-            {
-                // Note we are already inside the read, so checking for already going concurrent handshake.
-                _lockReadState = LockHandshake;
-            }
-
             await ForceAuthenticationAsync(adapter, receiveFirst: false, buffer).ConfigureAwait(false);
```

Comment:
This async method is no longer necessary, as it's just calling one method and awaiting it.  If you want to keep the method for some reason, remove "async" from its signature, and change the await to just be a return.  Otherwise, the callsites can just be changed to call directly the thing being awaited.

Reply:
see the comment above.